pkg/block/index.go

```diff
@@ -665,8 +670,16 @@ func rewrite(
 		return labels.Compare(series[i].lset, series[j].lset) < 0
 	})
 
+	lastSet := labels.Labels{}
 	// Build a new TSDB block.
 	for _, s := range series {
+		if labels.Compare(lastSet, s.lset) == 0 {
```

Comment:
The series are sorted based on their labelSet.  What happens if two labelSets are identical?  For one they will be consecutive in the slice.  So if the labelSet we have now is the same as the previous labelSet we have a duplicate series which the TSDB code will produce an error on.  So we just drop that series.

Reply:
Done.  Thank you!

Reply:
Sorry one more thing. As this shouldn't generally happen, let's add a metric so we can monitor when it does happen.

Reply:
The repair process batch job only which makes metrics less useful.  I had some similar comments in https://github.com/improbable-eng/thanos/pull/964

Perhaps I should leave a comment if that were to change?

Reply:
Yes let's in that case put a TODO in there

Reply:
TODO added.